jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -866,6 +908,36 @@ static void write(Sink sink, boolean last, String utf8Content, Callback callback
             sink.write(last, ByteBuffer.wrap(utf8Content.getBytes(StandardCharsets.UTF_8)), callback);
         }
 
+        static void write(Sink sink, boolean last, Content.Source source, Callback callback)
+        {
+            Content.Source.Aware aware = findContentSourceAware(sink);
+            if (aware != null)
+            {
+                // Optimization to enable zero-copy.
```

Comment:
Explain in the comment how the optimization still uses the sink.write path, so that commit logic etc. can be triggered.    Specifically, we do not do `aware.transfer(source, callback)` because we might need to commit the response.